### PR TITLE
Switch all Tools to net feed

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Publish all nuget tools to the azure-sdk-for-net DevOps feed using a single pipeline.

### CODE REVIEW

File: src/dotnet/Azure.ClientSdk.Analyzers/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
```
Comment (weshaggard) [inline src/dotnet/Azure.ClientSdk.Analyzers/ci.yml:1]: This can be a follow-up but we should probably move things from src/dotnet to tools directory.

**Review by weshaggard — APPROVED** [review]
The changes look good. Please make sure that all the pipelines run and work as expected.

File: .azure-pipelines/dotnet.yml [hunk 2]
```diff
@@ -1,93 +0,0 @@
-# External variables:
-# ProjectFile - The project to build and test. This variable is defined in pipeline web ui because we want to be able to provide it at queue time and that isn't supported in yaml yet.
-# VersioningProps - A collection of extra msbuild versioning properties like OfficialBuildId, PreReleaseVersionLabel, and DotNetFinalVersionKind
-
-trigger:
```
Comment (pakrym) [inline .azure-pipelines/dotnet.yml:5]: Why was this deleted?

File: src/dotnet/Azure.ClientSdk.Analyzers/ci.yml (hunk 1 above)
Comment (pakrym) [inline src/dotnet/Azure.ClientSdk.Analyzers/ci.yml:1]: This pipeline doesn't seem to be triggering.

Comment (chidozieononiwu) [reply src/dotnet/Azure.ClientSdk.Analyzers/ci.yml:1]: rEQUIRE pr cOMMENTS
Its probably because this option is checked.

Comment (weshaggard) [reply src/dotnet/Azure.ClientSdk.Analyzers/ci.yml:1]: @chidozieononiwu You didn't create a public pipeline for this ci.yml only an internal one. I renamed the old azure-sdk-tools - dotnet pipeline in public to tools - Azure.ClientSdk.Analyzers - ci to fix this. You might also need to setup a public pipeline for the snippet generator.
